### ISSUE TITLE
Compatibility issue between ara client version 1.6.0 and ara server version 1.5.8

### ISSUE THREAD

**Issue by hille721** [ansible-community/ara#467]
During an automic update of one ara client I recognized a compatibility issue between ara client version 1.6.0 and ara server 1.5.8. The playbook get's created in ara, but updating fails:

```
...
[WARNING]: Failure using method (v2_playbook_on_task_start) in callback plugin (<ansible.plugins.callback.ara_default.CallbackModule object at 0x7f3cafeb1610>): 'uuid'
Callback Exception: 
  File "/home/hille/gitrepos/plminfra_ansible/.venv/lib/python3.9/site-packages/ansible/executor/task_queue_manager.py", line 450, in send_callback
    method(*new_args, **kwargs)
   File "/home/hille/gitrepos/plminfra_ansible/.venv/lib/python3.9/site-packages/ara/plugins/callback/ara_default.py", line 504, in v2_playbook_on_task_start
    self._end_task()
   File "/home/hille/gitrepos/plminfra_ansible/.venv/lib/python3.9/site-packages/ara/plugins/callback/ara_default.py", line 602, in _end_task
    task_uuid = self.task["uuid"]
...

[WARNING]: Failure using method (v2_runner_on_ok) in callback plugin (<ansible.plugins.callback.ara_default.CallbackModule object at 0x7f3cafeb1610>): 'NoneType' object has no attribute 'submit'
Callback Exception: 
  File "/home/hille/gitrepos/plminfra_ansible/.venv/lib/python3.9/site-packages/ansible/executor/task_queue_manager.py", line 450, in send_callback
    method(*new_args, **kwargs)
   File "/home/hille/gitrepos/plminfra_ansible/.venv/lib/python3.9/site-packages/ara/plugins/callback/ara_default.py", line 534, in v2_runner_on_ok
    self._submit_thread("task", self._load_result, result, "ok", **kwargs)
   File "/home/hille/gitrepos/plminfra_ansible/.venv/lib/python3.9/site-packages/ara/plugins/callback/ara_default.py", line 359, in _submit_thread
    threads.submit(func, *args, **kwargs)
...

[WARNING]: Failure using method (v2_playbook_on_stats) in callback plugin (<ansible.plugins.callback.ara_default.CallbackModule object at 0x7f3cafeb1610>): 'uuid'
Callback Exception: 
  File "/home/hille/gitrepos/plminfra_ansible/.venv/lib/python3.9/site-packages/ansible/executor/task_queue_manager.py", line 450, in send_callback
    method(*new_args, **kwargs)
   File "/home/hille/gitrepos/plminfra_ansible/.venv/lib/python3.9/site-packages/ara/plugins/callback/ara_default.py", line 586, in v2_playbook_on_stats
    self._end_task()
   File "/home/hille/gitrepos/plminfra_ansible/.venv/lib/python3.9/site-packages/ara/plugins/callback/ara_default.py", line 602, in _end_task
    task_uuid = self.task["uuid"]
```

In ara the playbook will be then shown forever as running.

Good thing is, running ara server 1.6.0 and ara client 1.5.8 is working without any problems (of course some things introduced with 1.6.0 are then n/a in the GUI:
![image](https://user-images.githubusercontent.com/38468371/206169986-d85fcd41-fd87-47d4-9c6e-1efbe7366c21.png)

This issues should be more a tip for everyone, that the upgrade path to 1.6.0 should be:
1. upgrade server
2. upgrade all your clients

This ensures that ara is still working also in a transition phase.

@dmsimard : maybe you could also announce this somewhere.

**Comment by dmsimard** [comment]
@hille721 thanks for the issue, the explanation makes sense.

There is a mention of it in the upgrade notes:
![Screenshot from 2022-12-08 00-06-32](https://user-images.githubusercontent.com/1291204/206361534-624ce27d-6b72-4384-a800-60cfaab8a4fd.png)

We can do better next time, we don't currently have integration tests for forward and backwards compat but I could see that being useful.

I will leave this issue open for now.

**Comment by phemmer** [comment]
I just hit a similar issue (#490). I think it would be good for the images on quay & docker hub to be tagged with their specific version. I hit this issue because the quay image is out of date. I was expecting the "latest" tag to match the PIP package, but it doesn't. Even if "latest" was correct, it's still possible to have issues as I wouldn't be able to run an older version of the server, which I might need in case upgrading the clients is difficult or time consuming.

Basically since version matching is apparently important, we need to be able to ensure the correct version of the server.
